scripts/participateInAuction.js

```diff
@@ -55,6 +66,59 @@ async function participateInAuction(coordinatorAddress, network) {
   }
 }
 
+async function unlockMgnAndTokens({
+  pool1, pool2, pool1State, pool2State, infoAuction
+}) {
+  const depositToken = await pool1.depositToken()
+  const secondaryToken = await pool1.secondaryToken()
+
+  const dx1 = await DutchExchange.at(await pool1.dx())
+  const dx2 = await DutchExchange.at(await pool2.dx())
+
+  const lastAuction1 = (await pool1.lastParticipatedAuctionIndex()).toNumber()
+  const lastAuction2 = (await pool2.lastParticipatedAuctionIndex()).toNumber()
+
+  const dxAuctionIndex1 = (await dx1.getAuctionIndex.call(depositToken, secondaryToken)).toNumber()
+  const dxAuctionIndex2 = (await dx2.getAuctionIndex.call(secondaryToken, depositToken)).toNumber()
+
+  const logBeforeMsg = "Pool %d: The period is over. Unlocking MGN (transition to state DepositWithdrawnFromDx: 2)"
+  const logAfterMsg = "Pool %d: Successfully called triggerMGNunlockAndClaimTokens! in tx: %s"
+  if (pool1State === STATE_POOLING_ENDED && dxAuctionIndex1 > lastAuction1) {
+    infoAuction(logBeforeMsg, 1)
+    const result = await pool1.triggerMGNunlockAndClaimTokens()
+    infoAuction(logAfterMsg, 1, result.tx)
+  }
+  if (pool2State === STATE_POOLING_ENDED && dxAuctionIndex2 > lastAuction2) {
+    infoAuction(logBeforeMsg, 2)
+    const result = await pool2.triggerMGNunlockAndClaimTokens()
+    infoAuction(logAfterMsg, 2, result.tx)
+  }
+}
+
+
+async function claimMgn({
+  poolNumber, pool, state, debugAuction, infoAuction
+}) {
+  debugAuction("Waiting for the MGN unlock. Checking if we can claim the MGN")
+
+  // We could check the expire time easily of the MGN, but we will take the simple approach of
+  // executing "withdrawUnlockedMagnoliaFromDx" if the call succeeds
+
+  if (state === STATE_UNLOCKING_MGN) {
+    const canClaim = await pool.withdrawUnlockedMagnoliaFromDx.call()
+      .then(() => true)
+      .catch(() => false)
```

Comment:
Is there any way for us to tell if this failed for an unexpected reason? In this case we should probably alert the channel. Should we at least print the reason here?

It could also be a reminder for us to turn off the bot at the end of the pooling period.